File: ios/RecordToolManager.m
//
//  RecordToolManager.m
//  ReactNativeAppDemo
//
//  Created by JasonXD on 2019/5/25.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RecordToolManager, RCTViewManager)

  RCT_EXTERN_METHOD(startRecordFromManager:(nonnull NSNumber *)node)
  RCT_EXTERN_METHOD(stopRecordFromManager:(nonnull NSNumber *)node)

@end

File: ios/RecordToolManager.swift
//
//  RecordToolManager.swift
//  ReactNativeAppDemo
//
//  Created by JasonXD on 2019/5/25.
//  Copyright © 2019 Facebook. All rights reserved.
//

@objc(RecordToolManager)
class RecordToolManager: RCTViewManager {
  
  override func view() -> UIView! {
    return RecordTool()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func startRecordFromManager(_ node: NSNumber) {
    
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view( forReactTag: node ) as! RecordTool
      component.startRecording()
    }
  }
  
  @objc func stopRecordFromManager(_ node: NSNumber) {
    
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view( forReactTag: node ) as! RecordTool
      component.stopRecording()
    }
  }
  
}
